File: 2073-time-needed-to-buy-tickets/2073-time-needed-to-buy-tickets.go
func timeRequiredToBuy(tickets []int, k int) int {
    n := len(tickets)
    if k < n-1 {
        ticketsAfterK := 0
        for i:=k+1;i<n;i++{
            ticketsAfterK += tickets[i]
        }
        tickets = tickets[0:k+1]
        tickets = append(tickets, ticketsAfterK)
    }

    if k > 0 {
        ticketsBeforeK := 0
        for i:=0;i<k;i++{
            ticketsBeforeK += tickets[i]
        }
        tickets = tickets[k:]
        tickets = append([]int{ticketsBeforeK}, tickets...)
    }
    fmt.Println("after", tickets)

    kIdx := k
    time := 0
    tmp := -1
    for len(tickets) > 0 {
        if tickets[0] > 0 {
            time += 1
            tickets[0] -= 1
            
                tmp = tickets[0]
                
            if tickets[0] == 0 {
                tickets = tickets[1:]
            }else{
                tickets = append(tickets, tmp)
            }
        }else{

        }
    }
    return time
}

File: 2073-time-needed-to-buy-tickets/2073-time-needed-to-buy-tickets.rs
impl Solution {
    pub fn time_required_to_buy(tickets: Vec<i32>, k: i32) -> i32 {
        let mut people: Vec<[i32;2]> = Vec::new();

        for (idx , &el) in tickets.iter().enumerate() {
            if idx == k as usize {
                people.push([el, 1]);
            }else {
                people.push([el, 0]);   
            }
        }
        let mut temp: [i32;2];
        let mut time = 0;
        while people.len() > 0 {
            time += 1;
            if people[0][0] - 1 == 0 {
                if people[0][1] == 1 {
                    break
                }else{
                    people.remove(0);
                }
            }else {
                temp = people[0];
                temp[0] -= 1;

                people.remove(0);
                people.push(temp);
            }
        }

        return time
    }
}